### PR TITLE
fix: ColorPicker hover boundary issues

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
#42525 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix ColorPicker hover boundary issue      |
| 🇨🇳 Chinese |     修复颜色选择器悬停边界问题      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a6abad9</samp>

This pull request enhances the `ColorPicker` component by adding a test case, a ref, and a new prop to improve the popup behavior and the color selection feedback. It also updates the `@rc-component/color-picker` dependency and formats `package.json`.

### <samp>🤖 Generated by Copilot at a6abad9</samp>

*  Update `@rc-component/color-picker` dependency version to `~1.2.0` ([link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L117-R117))
*  Add `popupAllowCloseRef` ref to control popup closing behavior based on color change source ([link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL8-R8), [link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL100-R102), [link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR116-R119), [link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR127-R130), [link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL152-R172))
*  Add `onChangeComplete` prop and function to `ColorPickerPanel` and `ColorPickerBase` components to reset `popupAllowCloseRef` when color change is completed ([link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-3abf1a34a643a7087dfbaf3039447778574de2b61e627f4c1218297fb961c70fL13-R28), [link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-3abf1a34a643a7087dfbaf3039447778574de2b61e627f4c1218297fb961c70fL48-R60), [link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR127-R130), [link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL152-R172))
*  Import `createEvent` and `spyElementPrototypes` functions to create custom mouse events and mock element position and size for testing ([link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4L1-R2))
*  Add `doMouseMove` function to simulate mouse movement and events for dragging color picker handler in test cases ([link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4R10-R31))
*  Add test case for checking hover boundary issues using `spyElementPrototypes` and `doMouseMove` functions ([link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-aa06460351e8630f305cf6509e8137984e0f167ab2560da22874033189234fb4R283-R301))
*  Add newline character at the end of `package.json` file ([link](https://github.com/ant-design/ant-design/pull/42669/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L327-L326))

### CODE REVIEW

**Comment by afc163** [comment]
怎么大这么多

**Comment by RedJue** [comment]
> 怎么大这么多

是漏合了什么么，代码没写多少。

**Comment by yoyo837** [comment]
> > 怎么大这么多
> 
> 是漏合了什么么，代码没写多少。

会不会是color-picker@1.2.0带来的体积？

**Comment by RedJue** [comment]
> > > 怎么大这么多
> 
> > 
> 
> > 是漏合了什么么，代码没写多少。
> 
> 
> 
> 会不会是color-picker@1.2.0带来的体积？

应该也不会，就加了几个API，也没几行代码🤔

**Comment by RedJue** [comment]
> > > 怎么大这么多
> 
> > 
> 
> > 是漏合了什么么，代码没写多少。
> 
> 
> 
> 会不会是color-picker@1.2.0带来的体积？

可能是什么依赖更新了，引了大体积的库

File: package.json [hunk 1]
```diff
@@ -114,7 +114,7 @@
     "@ant-design/react-slick": "~1.0.0",
     "@babel/runtime": "^7.18.3",
     "@ctrl/tinycolor": "^3.6.0",
-    "@rc-component/color-picker": "~1.1.1",
+    "@rc-component/color-picker": "~1.2.0",
```
Comment (li-jia-nan) [inline package.json:117]: 是不是直接用 ^ 就行了

Comment (RedJue) [reply package.json:117]: 应该可以了，修复版本 master 已经发过了，应该跟着升就行。

Comment (MadCcc) [reply package.json]: 用 `~`，不然 rc 的 breaking change 只能发 major 了

File: package.json [hunk 2]
```diff
@@ -114,7 +114,7 @@
     "@ant-design/react-slick": "~1.0.0",
     "@babel/runtime": "^7.18.3",
     "@ctrl/tinycolor": "^3.6.0",
-    "@rc-component/color-picker": "~1.1.1",
+    "@rc-component/color-picker": "^1.2.0",
```
Comment (MadCcc) [inline package.json:117]:
```suggestion
    "@rc-component/color-picker": "~1.2.0",
```

Comment (RedJue) [reply package.json:117]: 不需要跟着 `minor` 走么？

Comment (yoyo837) [reply package.json:117]: 走手动feature更新，和其他rc-*一样做法

Comment (RedJue) [reply package.json:117]: ok，那就这样处理

**Review by MadCcc — APPROVED** [review]
